crown-court-contribution/src/main/java/uk/gov/justice/laa/crime/contribution/service/MaatCalculateContributionService.java

```diff
@@ -304,9 +308,20 @@ annualDisposableIncome, isUpliftApplied(calculateContributionDTO,
                         calculateContributionDTO.getContributionCap()
                 );
 
-        // Revisit the request to pass the offenceType object for Contribs Cap
-        ApiCalculateContributionResponse apiCalculateContributionResponse =
-                calculateContributionService.calculateContribution(apiCalculateContributionRequest);
+        Optional<ApiAssessment> fullAssessment = calculateContributionDTO.getAssessments().stream()
+                .filter(it -> it.getAssessmentType() == AssessmentType.FULL).findFirst();
+        String fullResult = fullAssessment.map(assessment -> assessment.getResult().name()).orElse(null);
+
+        ApiCalculateContributionResponse apiCalculateContributionResponse = null;
+        if (Constants.INEL.equals(fullResult)) {
+            apiCalculateContributionResponse = getApiCalculateContributionResponse();
+            totalMonths = 0;
+        } else {
+            // Revisit the request to pass the offenceType object for Contribs Cap
+            apiCalculateContributionResponse =
+                    calculateContributionService.calculateContribution(apiCalculateContributionRequest);
+        }
+
```

Comment:
Are we sure this is the right place for this additional logic? I thought we'd provisionally agreed to update the `performContributions` method either by adding an extra branch or reversing the comparison e.g.

```java
if (Constants.Y.equals(contributionResponseDTO.getCalcContribs()) ||
        contributionResponseDTO.getId() != null ||
        (calculateContributionDTO.getMonthlyContributions() != null && 
        calculateContributionDTO.getMonthlyContributions()
            .compareTo(BigDecimal.ZERO) > 0) || !Constants.INEL.equals(fullResult)) {
    result = calculateContributions(calculateContributionDTO, contributionResponseDTO);
} else {
     result = ContributionResult.builder()
         .monthlyAmount(BigDecimal.ZERO)
         .contributionCap(BigDecimal.ZERO)
         .upfrontAmount(BigDecimal.ZERO)
         .build();
``` 

I'm not sure why we aren't setting the `totalMonths` in the response, I think we probably should be.

Reply:
That's how I tried first, but the effective date calculation happens inside the calculateContributions method, which is a mandatory field and the create contribution fails. So, referred to the stored procedure and kept it similar to how it is implemented in the SP and it works now.
Also, we are setting the totalMonths. please refer to line no 318 & 331
Tested this by running the functional tests in Test and all the contribution test cases passed

Reply:
Could we set the `effectiveDate` in the `else` block and then call the same method? This would avoid duplicating the `INEL` logic across two branches and executing unnecessary code just to determine the value of a single field.


```java
if (Constants.Y.equals(contributionResponseDTO.getCalcContribs())
    || contributionResponseDTO.getId() != null
    || (calculateContributionDTO.getMonthlyContributions() != null
    && calculateContributionDTO.getMonthlyContributions().compareTo(BigDecimal.ZERO) > 0)
    || !Constants.INEL.equals(fullResult)) {
    result = calculateContributions(calculateContributionDTO, contributionResponseDTO);
} else {
    result = ContributionResult.builder()
        .monthlyAmount(BigDecimal.ZERO)
        .contributionCap(BigDecimal.ZERO)
        .upfrontAmount(BigDecimal.ZERO)
        .totalMonths(0)
        .effectiveDate(getEffectiveDate(calculateContributionDTO))
        .build();
}
```